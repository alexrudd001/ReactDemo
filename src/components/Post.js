import React from 'react';
import { Link } from 'react-router-dom';

const Post = ({ id, title, body }) => {
    return (
        <div key={id} className="post">
            <Link to={`/post/${id}`}>
                <h3>{title}</h3>
            </Link>
            <p>{body}</p>
        </div>
    );
};

export default Post;
