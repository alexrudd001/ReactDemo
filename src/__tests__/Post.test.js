import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Post from '../components/Post';

test('renders Post correctly', () => {
    const post = {
        id: 1,
        title: 'post title',
        body: 'post body',
        userId: 1
    };
    const wrapper = shallow(<Post key={1} {...post} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});
