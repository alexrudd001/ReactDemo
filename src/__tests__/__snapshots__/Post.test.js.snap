// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Post correctly 1`] = `
<div
  className="post"
  key="1"
>
  <Link
    to="/post/1"
  >
    <h3>
      post title
    </h3>
  </Link>
  <p>
    post body
  </p>
</div>
`;
